internal/bodyprocessors/json.go

```diff
@@ -41,28 +44,38 @@ func (js *jsonBodyProcessor) ProcessResponse(reader io.Reader, v plugintypes.Tra
 	return nil
 }
 
-func readJSON(reader io.Reader) (map[string]string, error) {
+func readJSON(reader io.Reader, maxRecursion int) (map[string]string, error) {
 	s := strings.Builder{}
 	_, err := io.Copy(&s, reader)
 	if err != nil {
 		return nil, err
 	}
 
-	json := gjson.Parse(s.String())
 	res := make(map[string]string)
 	key := []byte("json")
-	readItems(json, key, res)
-	return res, nil
+
+	if !gjson.Valid(s.String()) {
```

Comment:
do we really need to do this? the first impression is this is very expensive, the second one is when we use a low body limit this will always be invalid isn't/

Reply:
It is not very expensive. Of course, it is _more_ expensive than not doing vlaidation (adds roughly 33% more time in my benchmarks).

The upstream library [says](https://github.com/tidwall/gjson?tab=readme-ov-file#validate-json) that _If you are consuming JSON from an unpredictable source then you may want to validate prior to using GJSON._ I agree, for the case of a WAF.

Or we can forget about validation, and see if someone can bypass us. 🤷 If we do, we should document clearly this decision that we prefer speed over security here.